### PR TITLE
Employee profile setup

### PR DESCRIPTION
**Pull request by RupertSaxton**
New users are prompted to complete employee profile and add a business.  They can either create a new business or search through existing businesses.  Added links to customise their profile.

### CODE REVIEW

File: skedulr/app/models/business.rb [hunk 1]
```diff
@@ -1,3 +1,11 @@
 class Business < ApplicationRecord
   has_many :employees
+
+  def self.search(term)
+    if term
+      where('name LIKE ?', "%#{term}%")
```
Comment (evankozliner) [inline skedulr/app/models/business.rb]: This search is a good start, one slight tweak: After you've made sure you're lowercasing in the before_save method be sure to lowercase the search term here. Later we can call this search with ajax

File: skedulr/app/models/business.rb [hunk 2]
```diff
@@ -1,3 +1,11 @@
 class Business < ApplicationRecord
   has_many :employees
+
```
Comment (evankozliner) [inline skedulr/app/models/business.rb:3]: Could you add a before_save here to lower case the business name before it's put into the database?

File: skedulr/app/views/businesses/index.html.erb [hunk 3]
```diff
@@ -1 +1,13 @@
...
+<%= link_to "Create New Business", new_business_path %> or
+<p> Search for existing business </p>
+<%= form_tag(businesses_path, method: :get) do %>
+  <%= text_field_tag :term, params[:term] %>
+  <%= submit_tag 'Search', name: nil %>
+<% end %>
+<% for business in @businesses do %>
+  <ul>
+    <li>
+      <%= link_to business.name, employee_path(@employee, employee: {business_id: business.id}), method: :put %>
```
Comment (evankozliner) [inline skedulr/app/views/businesses/index.html.erb:10]: Display the business name With Each Word Capitalized (because we will lowercase them in the db)

**Comment by evankozliner** [comment]
Good start! Soon we'll need to scrap "is_manager" from the Employee model and add a table of managers ids but that doesn't have to be in this request.

File: skedulr/app/models/business.rb (hunk 2 above)
Comment (RupertSaxton) [reply skedulr/app/models/business.rb:3]: Done!

File: skedulr/app/models/business.rb (hunk 1 above)
Comment (RupertSaxton) [reply skedulr/app/models/business.rb]: Done!

File: skedulr/app/views/businesses/index.html.erb (hunk 3 above)
Comment (RupertSaxton) [reply skedulr/app/views/businesses/index.html.erb:10]: I called after_find in the business controller to do it as we reference the business name in multiple views so defining it in one place helps coherency

**Pull request closed without merging**